Tools/scripts/build_binaries_history.py

```diff
@@ -87,3 +87,66 @@ def record_run(self, hash, tag, start_time, duration):
                   "values (?, ?, ?, ?)",
                   (hash, tag, start_time, duration))
         c.commit()
+
+    def generate_firmware_size_json(self, json_filepath, githash):
+        c = self.conn()
+        sql_query = "SELECT * FROM build WHERE hash = ?"
+        sql_result = c.execute(sql_query, (githash,))
+        records = sql_result.fetchall()
+        # print("Total rows are:  ", len(records))
+        c.close()
+        sql_dict = {
+            "hash": 0,
+            "tag": 1,
+            "vehicle": 2,
+            "board": 3,
+            "frame": 4,
+            "text": 5,
+            "data": 6,
+            "bss": 7,
+            "start_time": 8,
+            "duration": 9,
+        }
+        # check file exist to prevent json failure
+        with open(json_filepath, "a") as summary_json:
+            pass
+        import json
+        for row in records:
+            vehicle = row[sql_dict["vehicle"]]
+            board = row[sql_dict["board"]]
+            frame = row[sql_dict["frame"]]
+            text = row[sql_dict["text"]]
+            data = row[sql_dict["data"]]
+            bss = row[sql_dict["bss"]]
+
+            if vehicle == "Rover":
+                vehicle = "ardurover"  # match binary name on build/board/bin
+            if frame is not None:
+                vehicle = vehicle + "-" + frame
+            build_json = [{
+                board.lower(): {vehicle.lower(): {"text": text, "data": data, "bss": bss}},
+            }]
+            main_data_summary = build_json
+            # we need to check the file isn't empty
+            if os.path.getsize(json_filepath) > 0:
```

Comment:
Haven't we ensured this file doesn't exist in the caller?

Reply:
The json parser need a non empty file so we need to check that

Reply:
Yes.  So why did you remove that file in the caller?

Reply:
we need a fresh file to not carry previous build.